### ISSUE TITLE
`use_super_parameters` doesn't trigger if param is used in constructor

### ISSUE THREAD

**Issue by goderbauer** [dart-lang/linter#3613]
```dart
class Foo {
  Foo(this.foo);
  
  String foo;
}

class Bar extends Foo {
  Bar(String foo) : super(foo) { // 🔥 no lint warning is given
    print(foo);
  }
}
```

The code above should have a `use_super_parameters` because super parameters can be used here as shown below. However, no warning is given. If you comment out the print line, a warning is produced.

```dart
class Foo {
  Foo(this.foo);
  
  String foo;
}

class Bar extends Foo {
  Bar(super.foo) {
    print(foo);
  }
}
```

**Comment by goderbauer** [comment]
Actually, not producing the lint warning here is more correct because the `foo` getter in the base class could be unrelated to the `foo` parameter....

**Comment by natebosch** [comment]
I agree that this is a true negative. I don't think it's worth the effort to do the extra analysis of the super class to check whether the reference is a final field, unless that implementation is very easy.
